logstash-core/lib/logstash/json.rb

```diff
@@ -51,7 +51,12 @@ def jruby_dump(o, options = {})
     def normalize_encoding(data)
       case data
       when String
-        LogStash::UnicodeNormalizer.normalize_string_encoding(data)
```

Comment:
we _may_ be able to delete `LogStash::UnicodeNormalizer`, as I believe it is no longer referenced.

We don't need a normalized copy of the string; we simply need to emit a string that _can_ be normalized by JrJackson. In most cases this is simply the string that we received (no copy necessary), but for strings that are flagged as `BINARY`, we need to return a copy that has been flagged as utf-8 _so that_ the `RubyString#asJavaString()` that jrjackson uses will preserve any valid utf-8 sequences that it contains.

Reply:
- we are free to remove `LogStash::UnicodeNormalizer` as it is not referenced anywhere
- we developed unicode handling when JrJackson was v0.4.18 and with your `RubyUtils#writeRubyString` changes in in 0.4.20, you are right that we no longer need `String#encode & scrub!` _expensive_ operations

LGTM if you please remove `LogStash::UnicodeNormalizer`.